crates/control/src/behavior/dribble.rs

```diff
@@ -72,10 +81,37 @@ pub fn execute(
 fn is_kick_pose_reached(
     kick_pose_to_robot: Pose<Ground>,
     kick_info: &InWalkKickInfoParameters,
+    precision_kick: bool,
 ) -> bool {
-    let is_x_reached = kick_pose_to_robot.position().x().abs() < kick_info.reached_thresholds.x;
-    let is_y_reached = kick_pose_to_robot.position().y().abs() < kick_info.reached_thresholds.y;
-    let is_orientation_reached =
-        kick_pose_to_robot.orientation().angle().abs() < kick_info.reached_thresholds.z;
+    let thresholds = if precision_kick {
+        kick_info.precision_kick_reached_thresholds
+    } else {
+        kick_info.reached_thresholds
+    };
+
+    let is_x_reached = kick_pose_to_robot.position().x().abs() < thresholds.x;
+    let is_y_reached = kick_pose_to_robot.position().y().abs() < thresholds.y;
+    let is_orientation_reached = kick_pose_to_robot.orientation().angle().abs() < thresholds.z;
+
     is_x_reached && is_y_reached && is_orientation_reached
 }
+
+pub fn precision_kick(game_controller_state: Option<FilteredGameControllerState>) -> bool {
+    let game_controller_state = game_controller_state.unwrap_or_default();
+
+    let penalty_shootout = matches!(
+        game_controller_state.game_phase,
+        GamePhase::PenaltyShootout { .. }
+    );
+    let own_kick_off = matches!(
+        game_controller_state.game_state,
+        FilteredGameState::Playing {
+            kick_off: true,
+            ball_is_free: true
+        }
+    );
+    let sub_state = game_controller_state.sub_state.is_some();
+    let kicking = matches!(game_controller_state.kicking_team, Team::Hulks);
```

Comment:
I really think we should add a timeout to switch strategies. It would be very unfortunate if we verjockeln a goalkick and kassieren a goal then.

Reply:
Hmm, not sure if this is necessary, I suggest we just test it. I never had the feeling that the robot really needed more time to reach it's kick position.

Reply:
We testes this last testgame. I think you have to either tune parameters or implement the timeout strategy